tools/checkdocs/checkdocs.go

```diff
@@ -220,3 +238,89 @@ func verifyTags(fm []byte) error {
 
 	return nil
 }
+
+// checkTableFile verifies that supported-commands.md is correctly formatted,
+// using logf for progress reporting and fatalf for errors.
+func checkTableFile(file string, logf, fatalf func(string, ...any)) error {
+	fileInBytes, err := os.ReadFile(file)
+	if err != nil {
+		return fmt.Errorf("couldn't read file %s: %s", file, err)
+	}
+
+	verifyIssues(fileInBytes, logf, fatalf)
+
+	return nil
+}
+
+// verifyIssues checks that listed issues statuses.
+func verifyIssues(fm []byte, logf, fatalf func(string, ...any)) {
+	p, err := github.CacheFilePath()
+	if err != nil {
+		log.Panic(err)
+	}
+
+	clientDebugF := gh.NoopPrintf
+
+	client, err := github.NewClient(p, log.Printf, logf, clientDebugF)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	s := bufio.NewScanner(bytes.NewReader(fm))
+	for s.Scan() {
+		line := s.Text()
+
+		match := issueRE.FindStringSubmatch(line)
+		if len(match) == 0 {
+			continue
+		}
+
+		expectedMatchLen := 6
+
+		if len(match) != expectedMatchLen {
+			logf("invalid [issue]({URL}) format: %s", line)
+			continue
+		}
+
+		url := match[3]
+		repo := match[4]
+
+		var num int
+
+		num, err = strconv.Atoi(match[5])
+		if err != nil {
+			fatalf(err.Error())
+		}
+
+		if num <= 0 {
+			logf("invalid [issue]({URL}) incorrect issue number")
+			continue
+		}
+
+		if client == nil {
+			continue
+		}
+
+		var status github.IssueStatus
+
+		status, err = client.IssueStatus(context.TODO(), url, repo, num)
+		if err != nil {
+			fatalf(err.Error())
+		}
+
+		switch status {
+		case github.IssueOpen:
+			// nothing
+		case github.IssueClosed:
+			logf("invalid [issue]({URL}) linked issue %s is closed", url)
+		case github.IssueNotFound:
+			logf("invalid [issue]({URL}) linked issue %s is not found", url)
```

Comment:
I think these should be `log.Printf` to keep the consistency with log.Panicf below :hugs: